### DISCUSSION TITLE
ref conditional expression and increment operator

### DISCUSSION THREAD

**Discussion by FaustVX** [dotnet/csharplang#4227] [Language Ideas] [answered]
With the introduction of ref conditional, I thought this was possible :
```csharp
(ref a <= b ? ref x : ref y)++;
```

But I have to do that instead :
```csharp
ref var min = ref a <= b ? ref x : ref y;
min++;
```

And of course, it should valid for `--`, `+=`, …

**Comment by VolodymyrBS** [comment, marked as answer]
you already could do it but you should remove 'ref' keyword before 'a'.
take a look https://sharplab.io/#v2:D4AQTAjAsAUCDMACciDCiDetE+UkALIgLIAUAlNrljLnYgJYB2ALogIaIC8iYANIgBG3RPAEAPEQAYBAT2kBuKvUSlOAHh7CA/IgBOAUwBmiSQC59xxLPIBqW0tq4AvoljOgA===

**Reply by FaustVX** [reply]
Oh, ok.
I really thought that you should put ref at the start of ref conditional.
Thanks.
